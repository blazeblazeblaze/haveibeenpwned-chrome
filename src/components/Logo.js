import React from 'react';
import PropTypes from 'prop-types';
import Link from './Link';

const Logo = ({onStartOver}) => {
  return (
    <Link className="logotype" onClick={onStartOver}>
      <span className="logo-wrapper">';--</span>
    </Link>
  );
}

Logo.propTypes = {
  onStartOver: PropTypes.func.isRequired
};

export default Logo;
