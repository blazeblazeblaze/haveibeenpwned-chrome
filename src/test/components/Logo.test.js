import React from 'react';
import ReactDOM from 'react-dom';
import Logo from '../../components/Logo';
import renderer from 'react-test-renderer';

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Logo onStartOver={jest.fn} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('Logo renders correctly', () => {
  const component = renderer.create(
   <Logo onStartOver={jest.fn} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
