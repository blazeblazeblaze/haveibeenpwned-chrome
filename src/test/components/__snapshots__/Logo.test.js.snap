// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders correctly 1`] = `
<a
  className="logotype"
  href=""
  onClick={[Function]}
>
  <span
    className="logo-wrapper"
  >
    ';--
  </span>
</a>
`;
